### PR TITLE
Update styling of individual marker

### PR DESCRIPTION
**Pull request by taysea**
Figma: https://www.figma.com/file/bMVkdgHkimCENGHiKPGcYH/Maps?type=design&node-id=459-66511

Updates the individual marker styling to align with Figma designs and allow status specific renderings.

Approach:
- Implemented as styled Box with icon child. Open to feedback on direction.

The pros/cons below can guide discussions in terms of "ease of use" for developers/how we want to distribute

Pros:
- Allows theme to drive color styling
- Don't need to maintain specific icons for this purpose
- Makes it easy to change styling/direction in future

Cons:
- Had to do some `style` tag/styled component to rotate the box so the pointed part faced down
- The pin is not perfectly the same shape as the vector in Figma (would need to get fancier with CSS)
- Had to copy over some grommet utils to manage the pad + border calculations to keep the pins the same size even though critical has a bigger border

Screen Shot 2023-06-07 at 11 35 03 AM

https://github.com/grommet/grommet-leaflet/assets/12522275/170969fc-404c-4991-82ee-ee044446ab4d

Related to https://github.com/grommet/hpe-design-system/issues/3393

### CODE REVIEW

**Review by jcfilben — APPROVED** [review]
Nice work, I think the styled box with icon as a child seems like a good approach.

Just out of curiosity I tested the markers with a screen reader and they were being read out pretty well.

**Comment by jcfilben** [comment]
Just an additional note on this, I think for now sticking with the Box + Icon for the marker makes sense for a MVP. Eventually we could create new icons that match what we have on Figma (so we wouldn't have a Box with an Icon child, but instead just an Icon) so we don't have to do the style rotation on the triangle.

File: src/GrommetMarker.jsx [hunk 1]
```diff
@@ -1,21 +1,76 @@
...
+      border={border}
+      elevation="medium"
+      flex={false}
+      pad={adjustPadStyle(theme.global.edgeSize.xsmall, theme.global.borderSize[border.size] || border.size)}
+      round
+      theme={theme}
+      width={{min: 'fit-content'}}
+    >
+      {/* style needed to offset for rotation of location pin so triangle is vertical still */}
+      <StatusIcon color={statuses[status].color} size="small" style={status === 'warning' ? { transform: 'rotate(-45deg)' } : undefined} />
```
Comment (halocline) [inline src/GrommetMarker.jsx:60]: Would using Box's background object be an alternative? `rotate` is a property on that object.

Do we lose anything if the icon is not a child?

**Review by halocline — COMMENTED** [review]
Looking like a great start.

File: src/GrommetMarker.jsx [hunk 2]
```diff
@@ -1,21 +1,76 @@
...
+    color: 'status-unknown',
+    icon: StatusUnknownSmall
+  }
+}
+
+const StatusIcon = ({ status, theme }) => {
+  const border = { color: statuses[status].color, size: statuses[status].borderSize || 'small'};
+  const StatusIcon = statuses[status].icon;
+
+  return (
```
[the remark this answers is not in the document]
Comment (taysea) [reply src/GrommetMarker.jsx]: Good feedback. Done.

File: src/GrommetMarker.jsx [hunk 3]
```diff
@@ -1,21 +1,76 @@
...
+  // to create the pin shape
+  border-bottom-right-radius: 0;
+  transform: rotate(45deg);
+  &:hover {
+    // TO DO revisit "hover" color with designers. Use of "background-layer-overlay" feels strange semantically
+    border: ${props => `${props.theme?.global?.borderSize?.[props.border.size] || props.border.size} solid ${props.theme?.global?.colors?.['background-layer-overlay']}`};
+  }
+`;
+
+const statuses = {
```
[the remark this answers is not in the document]
Comment (taysea) [reply src/GrommetMarker.jsx]: Done.

**Review by halocline — COMMENTED** [review]
looks good. Two small comments.

File: src/GrommetGeoJSON.jsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import L from 'leaflet'
+import { Pin } from './markers';
+
+const GrommetGeoJSON = ({ data, ...rest }) => {
+
+    const pointToLayer = (feature, latlng) => L.marker(latlng, {
+        icon: L.divIcon({
+        // 'grommet-marker' class prevents leaflet default divIcon styles
+        className: 'grommet-marker',
+        html: ReactDOMServer.renderToString(<Pin status={feature?.grommet?.status} />),
```
[the remark this answers is not in the document]
Comment (taysea) [reply src/GrommetGeoJSON.jsx:13]: I'm leaning towards keep as is until requirements desire it.

File: src/GrommetGeoJSON.jsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        })
+    })
+
+    return (
+        <GeoJSON key="my-geojson" data={data} pointToLayer={pointToLayer} {...rest} />
```
[the remark this answers is not in the document]
Comment (taysea) [reply src/GrommetGeoJSON.jsx:18]: Agree to letting "rest" handle.

File: src/markers/Pin.jsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+      >
+        {/* style needed to offset for rotation of location pin so triangle is vertical still */}
+        {/* TO DO revisit icon sizes with designers */}
+        <StatusIcon color={STATUS_MAP[status].color} size="13px" style={status === 'warning' || status === 'unknown' ? { transform: 'rotate(-45deg)' } : undefined} />
+      </StyledBox>
+    </Grommet>
+  );
+}
+
+export { Pin };
```
Comment (taysea) [inline src/markers/Pin.jsx:44]:
```suggestion
export { Pin };

```

File: src/GrommetGeoJSON.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (halocline) [reply src/GrommetGeoJSON.jsx:13]: sounds good.

**Review by halocline — APPROVED** [review]

**Pull request merged**